.github/workflows/swift-toolchain.yml

```diff
@@ -939,55 +948,85 @@ jobs:
+            $PYTHON_INCLUDE_DIR = "${env:PYTHON_LOCATION_${{ matrix.arch }}}/include"
+            $ExeSuffix = ".exe"
+            Remove-Item env:\SDKROOT
+          } elseif ( "${{ matrix.os }}" -eq "Darwin" ) {
+            # Default swiftc comes from /usr/bin and is not compatible with the toolchain.
+            $CLANG_LOCATION = "${env:HOME}/Library/Developer/Toolchains/swift-6.0.1-RELEASE.xctoolchain/usr/bin"
+            $SWIFTC = Join-Path $CLANG_LOCATION "swiftc"
+            # We need to use llvm-17 to build the compiler on macOS. We get it from the Swift toolchain.
+            $CC = Join-Path $CLANG_LOCATION "clang"
+            $CXX = Join-Path $CLANG_LOCATION "clang++"
+            $CACHE = "${{ github.workspace }}/SourceCache/swift/cmake/caches/Darwin-${{ matrix.arch }}.cmake"
+            $SDKROOT = xcrun --sdk macosx --show-sdk-path
+            $EXTRA_FLAGS = "${{ matrix.extra_flags }}".Split()
+            $EXTRA_FLAGS += @("-D", "CMAKE_SYSTEM_NAME=Darwin")
+            $LIBPYTHON_PATH = "${env:pythonLocation}/lib/python3.9/config-3.9-darwin/libpython3.9.a"
+            $PYTHON_INCLUDE_DIR = "${env:pythonLocation}/include/python3.9"
+            $ExeSuffix = ""
           }
-          $SWIFTC = cygpath -m (Get-Command swiftc).Source
-          $SDKROOT = cygpath -m ${env:SDKROOT}
-          # Use toolchain clang to avoid broken __prefetch intrinsic on arm64 in Clang 18.
-          $CLANG_LOCATION = cygpath -m (Split-Path (Get-Command swiftc).Source)
-          Remove-Item env:\SDKROOT
+          $ClangTablegen = "${{ github.workspace }}/BinaryCache/0/bin/clang-tblgen${ExeSuffix}"
+          $ClangTidyConfusableCharsGen = "${{ github.workspace }}/BinaryCache/0/bin/clang-tidy-confusable-chars-gen${ExeSuffix}"
+          $LldbTablegen = "${{ github.workspace }}/BinaryCache/0/bin/lldb-tblgen${ExeSuffix}"
+          $LlvmConfig = "${{ github.workspace }}/BinaryCache/0/bin/llvm-config${ExeSuffix}"
+          $LlvmTablegen = "${{ github.workspace }}/BinaryCache/0/bin/llvm-tblgen${ExeSuffix}"
           cmake -B ${{ github.workspace }}/BinaryCache/1 `
-                -C ${{ github.workspace }}/SourceCache/swift/cmake/caches/${CACHE} `
+                -C "${CACHE}" `
                 -D CMAKE_BUILD_TYPE=Release `
-                -D CMAKE_C_COMPILER=cl `
+                -D CMAKE_C_COMPILER="${CC}" `
                 -D CMAKE_C_COMPILER_LAUNCHER=sccache `
-                -D CMAKE_C_FLAGS="${{ inputs.WINDOWS_CMAKE_C_FLAGS }}" `
-                -D CMAKE_CXX_COMPILER=cl `
+                -D CMAKE_C_FLAGS="${{ matrix.cflags }}" `
+                -D CMAKE_CXX_COMPILER="${CXX}" `
                 -D CMAKE_CXX_COMPILER_LAUNCHER=sccache `
-                -D CMAKE_CXX_FLAGS="${{ inputs.WINDOWS_CMAKE_CXX_FLAGS }} -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH" `
-                -D CMAKE_MT=mt `
+                -D CMAKE_CXX_FLAGS="${{ matrix.cxxflags }} -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH" `
```

Comment:
Can we hoist `-D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH` up into the matrix configuration? This is Windows (Microsoft STL) specific.

Reply:
I figured it was fine to leave it since it's essentially unused on macOS but for consistency, I added it to the pre-step configuration instead.